fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java

```diff
@@ -814,25 +815,30 @@ private Void computeValuesNode(ExpressionContext context, List<ColumnRefOperator
 
     @Override
     public Void visitLogicalRepeat(LogicalRepeatOperator node, ExpressionContext context) {
-        return computeRepeatNode(context, node.getOutputGrouping());
+        return computeRepeatNode(context, node.getOutputGrouping(), node.getGroupingIds(), node.getRepeatColumnRef());
     }
 
     @Override
     public Void visitPhysicalRepeat(PhysicalRepeatOperator node, ExpressionContext context) {
-        return computeRepeatNode(context, node.getOutputGrouping());
+        return computeRepeatNode(context, node.getOutputGrouping(), node.getGroupingIds(), node.getRepeatColumnRef());
     }
 
-    private Void computeRepeatNode(ExpressionContext context, List<ColumnRefOperator> outputGrouping) {
+    private Void computeRepeatNode(ExpressionContext context, List<ColumnRefOperator> outputGrouping,
+                                   List<List<Long>> groupingIds, List<Set<ColumnRefOperator>> repeatColumnRef) {
         Preconditions.checkState(context.arity() == 1);
-
+        Preconditions.checkState(outputGrouping.size() == groupingIds.size());
         Statistics.Builder builder = Statistics.builder();
-        for (ColumnRefOperator columnRef : outputGrouping) {
-            builder.addColumnStatistic(columnRef, ColumnStatistic.unknown());
+        for (int index = 0; index < outputGrouping.size(); ++index) {
+            // calculate the column statistics for grouping
+            List<Long> groupingId = groupingIds.get(index);
+            builder.addColumnStatistic(outputGrouping.get(index),
+                    new ColumnStatistic(Collections.min(groupingId), Collections.max(groupingId), 0, 8,
```

Comment:
`Collections.min` and `Collections.max` should only call once

Reply:
There are outputGrouping’s column statistics calculation, the maximum and minimum values of the column statistics for each outputGrouping are the same if only call once